### PR TITLE
DSD-1146: Allow CheckboxGroup to update programmatically

### PR DESCRIPTION
**Pull request by EdwinGuzman**
Fixes JIRA ticket [DSD-1146](https://jira.nypl.org/browse/DSD-1146)

## This PR does the following:

- Allows the `CheckboxGroup` to programmatically update its internal `Checkbox`es through the `value` prop.
- This issue came up in the MLN app.

## How has this been tested?

Locally in Storybook.

## Accessibility concerns or updates

- N/A

### Checklist:

- [ ] I have updated the Storybook documentation accordingly.
- [ ] I have added relevant accessibility documentation for this pull request.
- [ ] All new and existing tests passed.

### Front End Review:

- [ ] Review the Vercel preview deployment once it is ready.

### CODE REVIEW

File: src/components/CheckboxGroup/CheckboxGroup.stories.mdx [hunk 1]
```diff
@@ -460,3 +465,125 @@ const onSubmit = () => {
...
+export function CheckboxGroupValuesUpdateExample() {
+  const [value, setValue] = useState([]);
+  const onClick1 = () => {
+    setValue(["art", "science", "math"]);
+  };
+  const onClick2 = () => {
+    setValue(["math", "music", "magic"]);
+  };
+  const onClick3 = () => {
+    setValue(["physics", "science", "chemisty"]);
```
Comment (jackiequach) [inline src/components/CheckboxGroup/CheckboxGroup.stories.mdx]: Typo: chemistry

File: src/components/CheckboxGroup/CheckboxGroup.stories.mdx [hunk 2]
```diff
@@ -460,3 +465,125 @@ const onSubmit = () => {
...
+      <span>Set values to:</span>
+      <ButtonGroup>
+        <Button id="btn1" onClick={onClick1}>
+          "art", "science", "math"
+        </Button>
+        <Button id="btn2" onClick={onClick2}>
+          "math", "music", "magic"
+        </Button>
+        <Button id="btn3" onClick={onClick3}>
+          "physics", "science", "chemisty"
```
Comment (jackiequach) [inline src/components/CheckboxGroup/CheckboxGroup.stories.mdx]: Typo: chemistry

File: src/components/CheckboxGroup/CheckboxGroup.stories.mdx [hunk 3]
```diff
@@ -460,3 +465,125 @@ const onSubmit = () => {
   const CheckboxGroupValue = CheckboxGroupRef.current.value;
 };
 ```
+
+### Programmatically Update
+
+Within the `CheckboxGroup`, individual `Checkbox`es can be updated programmatically
+through the `value` prop. Like the `defaultValue` prop, the `value` prop also
```
Comment (jackiequach) [inline src/components/CheckboxGroup/CheckboxGroup.stories.mdx]: wording choice: replace "Like the `defaultValue`..." with "Similar to the `defaultValue`"

File: src/components/CheckboxGroup/CheckboxGroup.stories.mdx [hunk 4]
```diff
@@ -460,3 +465,125 @@ const onSubmit = () => {
...
+accepts an array of strings.
+
+When passing selected values through the `value` prop, the consuming app is now
+controlling the state of the values. By default, the `CheckboxGroup` component
+returns the value(s) checked to the consuming app through the `onChange` callback.
+Note that the `defaultValue` prop is not necessary unless if there are values
```
Comment (jackiequach) [inline src/components/CheckboxGroup/CheckboxGroup.stories.mdx]: remove "if"

File: src/components/CheckboxGroup/CheckboxGroup.stories.mdx [hunk 5]
```diff
@@ -460,3 +465,125 @@ const onSubmit = () => {
...
+that should be checked when the component first renders.
+
+In the following example, the `CheckboxGroup` component works as expected, but
+now there are three (3) additional buttons that can update the selected checkboxes
+within the `CheckboxGroup` component.
+
```
Comment (jackiequach) [inline src/components/CheckboxGroup/CheckboxGroup.stories.mdx:487]: Add a note letting users know they can see the selected values in the browser console.

**Review by jackiequach — COMMENTED** [review]
Cool feature! Just some doc comments.

**Comment by EdwinGuzman** [comment]
Great, thanks. Updated.

**Review by bigfishdesign13 — APPROVED** [review]
Looks good.

**Review by jackiequach — APPROVED** [review]

**Pull request merged**